airflow/www_rbac/views.py

```diff
@@ -2614,13 +2682,27 @@ def action_set_canceled(self, drs, session=None):
                     end_date=dr.execution_date,
                     include_subdags=True)
                 dagrun_count += 1
-        except Exception:
-            flash('Failed to set state to canceled', 'error')
+
+                # log this action
+                session.add(
+                    Log(
+                        'dagrun_list_set_canceled',
+                        task_instance=None,
+                        owner='anonymous' if g.user.is_anonymous else g.user.username,
+                        dag_id=dr.dag_id,
+                        execution_date=dr.execution_date,
+                        extra=str(list(request.values.items()))
+                    )
+                )
+        except Exception as e:
+            flash('Failed to set state to canceled: {}'.format(e), 'error')
         finally:
             flash(
                 "{count} dag runs and {altered_ti_count} task instances were canceled".format(
                     count=dagrun_count, altered_ti_count=task_instance_count))
-        return redirect(self.get_default_url())
+
+        self.update_redirect()
```

Comment:
same q for all of these. just answer one i guess.. what happens without it that you didnt like

Reply:
it would dump you back to the list page with none of your filters which was annoying. this way it keeps the filters you had set